ensure_tiller_installed.go

```diff
@@ -120,6 +122,91 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 		}
 	}
 
+	// Create the network policy for tiller so it is allowed to do its job in case all traffic is blocked.
+	{
+		networkPolicyName := tillerPodName
+		networkPolicyNamespace := c.tillerNamespace
+		protocolTCP := corev1.ProtocolTCP
+		tillerPort := intstr.IntOrString{
+			IntVal: 44134,
+		}
+		tillerHTTPPort := intstr.IntOrString{
+			IntVal: 44135,
+		}
+
+		name := fmt.Sprintf("%s/%s", networkPolicyNamespace, networkPolicyName)
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating networkpolicy %#q", name))
+
+		np := &networkingv1.NetworkPolicy{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "networking.k8s.io/v1",
+				Kind:       "NetworkPolicy",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      networkPolicyName,
+				Namespace: networkPolicyNamespace,
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app":  "helm",
+						"name": "tiller",
+					},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{
+					networkingv1.NetworkPolicyIngressRule{
+						Ports: []networkingv1.NetworkPolicyPort{
+							networkingv1.NetworkPolicyPort{
+								Protocol: &protocolTCP,
+								Port:     &tillerPort,
+							},
+							networkingv1.NetworkPolicyPort{
+								Protocol: &protocolTCP,
+								Port:     &tillerHTTPPort,
+							},
+						},
+					},
+				},
+				Egress: []networkingv1.NetworkPolicyEgressRule{
+					networkingv1.NetworkPolicyEgressRule{
+						To: []networkingv1.NetworkPolicyPeer{
+							networkingv1.NetworkPolicyPeer{
+								IPBlock: &networkingv1.IPBlock{
+									CIDR: "10.0.0.0/8",
```

Comment:
Is this the range for the nodes?
IIRC this tiller is installed in the tenant clusters and we have some clusters with different node IP ranges. Am I right?

Reply:
I'm not sure on the CIDRs but yes this is used to install tiller in control plane and tenant clusters. So we need to be able to support both.

Reply:
Those are reanges of private networks. It is not possible to restrict access in super granular way as we just don't have data about host networks and pod cidrs.

Anyway, this list should be changed to allow all the traffic as I've tested tiller after all the components were installed so I didn't catch external requests. Next commit fixes that